client/src/test/java/com/microsoft/durabletask/IntegrationTests.java

```diff
@@ -778,4 +788,54 @@ void purgeInstanceFilter() {
             assertFalse(metadata.isInstanceFound());
         }
     }
+
+    @Test()
+    void waitForInstanceStartThrowsException() {
+        final String orchestratorName = "orchestratorName";
+
+        DurableTaskGrpcWorker worker = this.createWorkerBuilder()
+                .addOrchestrator(orchestratorName, ctx -> {
+                    try {
+                        TimeUnit.SECONDS.sleep(5);
```

Comment:
It might be worth adding a comment here explaining why this test is correct.
```suggestion
                        // The orchestration remains in the "Pending" state until the first await statement
                        TimeUnit.SECONDS.sleep(5);
```